tools/shared.py

```diff
@@ -413,8 +413,14 @@ def sanity_is_correct():
       sanity_data = utils.read_file(sanity_file)
       if sanity_data == expected:
         logger.debug(f'sanity file up-to-date: {sanity_file}')
+        # Even if the sanity file is up-to-date we still need to at least
+        # check the llvm version. This comes at no extra performance cost
```

Comment:
This PR is much nicer now!

But I still don't think I get it. Why do we want to check this every time? That is stated here and in the PR description, but not explained. Are you worried about missing a problem and not warning about it? In what situation?

In general I don't see a reason to check this every time. It's important to check it when the config file changes, because then LLVM can change, but why otherwise?

Reply:
My expectation was that the warning would show ever time you run the compiler.  If folks don't want to see the warning they should run `-Wno-version-check`. 

The reason I found this bug is because i was experimenting with using `/usr/bin/clang` as my llvm backend.. and I was not seeing any warning when running the compiler.   

IIUC this was not supposed to be a warning that you just see just the first time you run the compiler, and I'm not aware of any warnings that have the property.  Not everyone is going to be paying attention the first time the compiler is run.. that invocation might even be hidden inside a cmake or configure test and then such as user might never see it.

Reply:
Is there some downside that you see to issuing this warning for each compiler run?

Reply:
I see now, thanks. Yes, I guess that is more consistent.